jvm-packages/xgboost4j/src/main/java/ml/dmlc/xgboost4j/java/XGBoost.java

```diff
@@ -187,11 +208,16 @@ public static Booster train(
       //evaluation
       if (evalMats.length > 0) {
         float[] metricsOut = new float[evalMats.length];
-        String evalInfo;
-        if (eval != null) {
-          evalInfo = booster.evalSet(evalMats, evalNames, eval, metricsOut);
+        String evalInfo = "";
+        if (evals != null && evals.length > 0) {
+          for (int i = 0; i < evals.length; i++) {
+            String evalLine = booster.evalSet(evalMats, evalNames, evals[i], metricsOut);
```

Comment:
isn't it replacing metricsOut for every time? 

and metricsOut is used in early stopping, etc. which makes it problematic

Reply:
Hmm, yeah. For early stopping, I'd probably need to add either an assumption about which eval metric is used, or a way for the user to choose which one matters. 

Are you fine with the first metric in the array of evaluations being used as the early stopping metric?

Reply:
Ok, disabled early stopping for multiple eval functions. I chose to simply print via rabit and continue training, but let me know if you'd prefer something else (namely a full error to be thrown).